src/test/java/org/jvnet/hudson/update_center/WarningsTest.java

```diff
@@ -0,0 +1,36 @@
+package org.jvnet.hudson.update_center;
+
+import junit.framework.Assert;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+
+import java.util.regex.Pattern;
+
+public class WarningsTest {
+    public void testValidJsonFile() throws Exception {
+        String warningsText = IOUtils.toString(this.getClass().getClassLoader().getResourceAsStream("warnings.json"));
+        JSONArray warnings = JSONArray.fromObject(warningsText);
+
+        for (int i = 0 ; i < warnings.size() ; i++) {
+            JSONObject o = warnings.getJSONObject(i);
+            assertNonEmptyString(o.getString("id"));
+            assertNonEmptyString(o.getString("type"));
+            assertNonEmptyString(o.getString("name"));
+            assertNonEmptyString(o.getString("message"));
+            assertNonEmptyString(o.getString("url"));
+            JSONArray versions = o.getJSONArray("versions");
+            for (int j = 0 ; j < versions.size() ; j++) {
+                JSONObject version = versions.getJSONObject(j);
+                String pattern = version.getString("pattern");
+                assertNonEmptyString(pattern);
+                Pattern p = Pattern.compile(pattern);
```

Comment:
nice3have: go through the list of plugin versions and ensure the expression returns true and false at least once